### PR TITLE
perf: move testing esm out of main snapshot

### PR DESCRIPTION
**Pull request by littledivy**
Closes https://github.com/denoland/deno/issues/21136

### CODE REVIEW

File: cli/js/40_testing.js [hunk 1]
```diff
@@ -3,1353 +3,1374 @@
...
-// Worker ops are an exception to this, because there is no way for the user to
-// await shutdown of the worker from the thread calling `worker.terminate()`.
-// Because of this, we give extra leeway for worker ops to complete, by waiting
-// for a whole millisecond if there are pending worker ops.
-function opSanitizerDelay(hasPendingWorkerOps) {
-  if (!hasSetOpSanitizerDelayMacrotask) {
-    core.setMacrotaskCallback(handleOpSanitizerDelayMacrotask);
-    hasSetOpSanitizerDelayMacrotask = true;
+// deno-lint-ignore-file
+{
```
Comment (bartlomieju) [inline cli/js/40_testing.js:7]: Why wrap it like this? It causes a huge diff for the reason I don't understand.

Comment (littledivy) [reply cli/js/40_testing.js:7]: Because `execute_script` runs this file in global scope unlike ESM and we don't want to leak the internal variables out.

Comment (bartlomieju) [reply cli/js/40_testing.js:7]: Oh right, I forgot that these are meant to be scripts, instead of modules.

File: cli/worker.rs [hunk 2]
```diff
@@ -610,7 +610,14 @@ impl CliMainWorkerFactory {
       options,
     );
 
-    if self.shared.subcommand.is_test_or_jupyter() {
+    if self.shared.subcommand.needs_test() {
+      worker.js_runtime.execute_script_static(
+        "40_testing.js",
```
Comment (bartlomieju) [inline cli/worker.rs:615]: Rename this to `ext:cli/js/40_testing.js` and the error from the LSP should be resolved. Please do the same for `40_jupyter.js` as well

File: cli/js/40_testing.js (hunk 1 above)
Comment (petamoriken) [inline cli/js/40_testing.js:7]: It might be better to have `"use strict";` with function scope.

Comment (littledivy) [reply cli/js/40_testing.js:7]: Same effect in both cases?

File: cli/tests/testdata/run/test_and_bench_in_run.js [hunk 3]
```diff
@@ -1,5 +0,0 @@
-Deno.test(function foo() {
-});
-
-Deno.bench(function bar() {
-});
```
Comment (bartlomieju) [inline cli/tests/testdata/run/test_and_bench_in_run.js:5]: Why remove this test?

Comment (littledivy) [reply cli/tests/testdata/run/test_and_bench_in_run.js:5]: because Deno.test is not available in `deno run`

Comment (bartlomieju) [reply cli/tests/testdata/run/test_and_bench_in_run.js:5]: `Deno.test` should be available in `deno run` but it should an empty function that does nothing. Similarly for `Deno.bench`. This was a regression test.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**